src/shaka_scheme/system/gc/GCData.hpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by mortrax on 2/8/18.
+//
+
+#ifndef SHAKA_SCHEME_GCDATA_HPP
+#define SHAKA_SCHEME_GCDATA_HPP
+
+namespace shaka {
+
+    /**
+     * @brief Implements the type of GCData, the internal
+     * manager object which holds the mark bit, determining
+     * when a data object is reclaimed by the Garbage Collector
+     */
+    class GCData {
+    public:
+        GCData(const Data& data);
+        ~GCData();
+
+        GCData *get_next();
```

Comment:
If I'm not mistaken, I thought we're not using `get_next()` here since it's a part of `GCList`? Or am I getting this wrong?